sdk/openai/openai/src/api/oaiSse.ts

```diff
@@ -10,29 +10,35 @@
  */
 
 import { StreamableMethod } from "@azure-rest/core-client";
-import { getSSEs } from "./getSSEs.js";
+import { getStream } from "./getSSEs.js";
 import { wrapError } from "./util.js";
+import { EventStream } from "../models/models.js";
+import { EventMessage, createSseStream } from "@azure/core-sse";
+import { polyfillStream } from "./readableStreamUtils.js";
 
-export async function* getOaiSSEs<TEvent, O extends Record<string, any>>(
+export async function getOaiSSEs<TEvent, O extends Record<string, any>>(
   response: StreamableMethod<unknown>,
   toEvent: (obj: O) => TEvent
-): AsyncIterable<TEvent> {
-  const stream = await getSSEs(response);
-  let isDone = false;
-  for await (const event of stream) {
-    if (isDone) {
-      // handle a case where the service sends excess stream
-      // data after the [DONE] event
-      continue;
-    } else if (event.data === "[DONE]") {
-      isDone = true;
-    } else {
-      yield toEvent(
-        wrapError(
-          () => JSON.parse(event.data),
-          "Error parsing an event. See 'cause' for more details"
+): Promise<EventStream<TEvent>> {
+  const stringStream = await getStream(response);
+  // TODO: use await using once we drop support for NodeJS v18.
+  const eventStream = createSseStream(stringStream);
+  const jsonParser = new TransformStream<EventMessage, TEvent>({
+    transform: async (chunk, controller) => {
+      if (chunk.data === "[DONE]") {
+        controller.terminate();
+        return eventStream[Symbol.asyncDispose]();
+      }
+      controller.enqueue(
+        toEvent(
+          wrapError(
+            () => JSON.parse(chunk.data),
+            "Error parsing an event. See 'cause' for more details"
+          )
         )
       );
-    }
-  }
+    },
+  });
+  /** TODO: remove these polyfills once all supported runtimes support them */
```

Comment:
Can't wait until we can drop these, I find the Chromium issue with ReadableStream not being async iterable particularly frustrating.